File: orion-ui/packages/orion-design/src/maps/stateDetails.ts
import { IStateDetailsResponse } from '@/models/IStateDetailsResponse'
import { IStateDetails } from '@/models/StateDetails'
import { MapFunction } from '@/services/Mapper'

export const mapIStateDetailsResponseToIStateDetails: MapFunction<IStateDetailsResponse, IStateDetails> = function(source: IStateDetailsResponse): IStateDetails {
  return {
    flowRunId: source.flow_run_id,
    taskRunId: source.task_run_id,
    childFlowRunId: source.child_flow_run_id,
    cacheKey: source.cache_key,
    scheduledTime: this.map('string', source.scheduled_time, 'Date'),
    cacheExpiration: this.map('string', source.cache_expiration, 'Date'),
  }
}

export const mapIStateDetailsToIStateDetailsResponse: MapFunction<IStateDetails, IStateDetailsResponse> = function(source: IStateDetails): IStateDetailsResponse {
  return {
    'flow_run_id': source.flowRunId,
    'task_run_id': source.taskRunId,
    'child_flow_run_id': source.childFlowRunId,
    'cache_key': source.cacheKey,
    'scheduled_time': this.map('Date', source.scheduledTime, 'string'),
    'cache_expiration': this.map('Date', source.cacheExpiration, 'string'),
  }
}

File: src/prefect/client/schemas/objects/state_details.py
from typing import (
    Dict,
    Optional,
)
from uuid import UUID

from pydantic_extra_types.pendulum_dt import DateTime

from prefect._internal.schemas.bases import PrefectBaseModel


class StateDetails(PrefectBaseModel):
    flow_run_id: Optional[UUID] = None
    task_run_id: Optional[UUID] = None
    # for task runs that represent subflows, the subflow's run ID
    child_flow_run_id: Optional[UUID] = None
    scheduled_time: Optional[DateTime] = None
    cache_key: Optional[str] = None
    cache_expiration: Optional[DateTime] = None
    deferred: Optional[bool] = None
    untrackable_result: bool = False
    pause_timeout: Optional[DateTime] = None
    pause_reschedule: bool = False
    pause_key: Optional[str] = None
    run_input_keyset: Optional[Dict[str, str]] = None
    refresh_cache: Optional[bool] = None
    retriable: Optional[bool] = None
    transition_id: Optional[UUID] = None
    task_parameters_id: Optional[UUID] = None